src/mbgl/map/tile_worker.cpp

```diff
@@ -65,22 +67,18 @@ TileParseResult TileWorker::parsePendingLayers() {
     // dependencies.
     for (auto it = pending.begin(); it != pending.end();) {
         auto& layer = *it->first;
-        auto& bucket = it->second;
+        auto bucket = dynamic_cast<SymbolBucket*>(it->second.get());
```

Comment:
Turns out there's not a good way to do a moving dynamic cast of a unique_ptr, so I'm going to comment the assert on the next line and leave it as-is.